runelite-client/src/main/java/net/runelite/client/plugins/bankvalue/BankValuePlugin.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018, TheLonelyDev <https://github.com/TheLonelyDev>
+ * Copyright (c) 2018, Jeremy Plsek <https://github.com/jplsek>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.bankvalue;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+@PluginDescriptor(name = "Bank Value")
+public class BankValuePlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private BankCalculation bankCalculation;
+
+	@Inject
+	private BankTitle bankTitle;
+
+	@Provides
+	BankValueConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(BankValueConfig.class);
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		bankTitle.reset();
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
```

Comment:
Have you tried hooking to the widget (un)hidden event for the bank instead?

Reply:
Yes, I tried it in the [previous commits](https://github.com/jplsek/runelite/blob/763adc1a6a7887b8f40994e155f097240aaea7b2/runelite-client/src/main/java/net/runelite/client/plugins/bankvalue/BankValuePlugin.java#L103) (if that's what your referring to). I removed it to avoid keeping track of a bankActive field.

Part of the problem is that the event gets called 1000's of times when opening the bank, and there doesn't seem to be a way to know when the bank is "done" loading. I tried to do it based on how many times the bank title widget gets updated, but it's a minimum of 4 and a maximum of how many tabs the player has. It falls apart when the user switches tabs/withdraw/deposit too quickly because only 2 events get fired, before calling 4 more. The code gets really ugly.

The WidgetHiddenChanged event can be put back, but calculating the bank using the async method that was added still needs to be done in the game tick (since the first call is usually null, so a second call needs to be made eventually). It would look the same as what was linked.

I would like to think that I'm just missing something really obvious, but I'm not sure.